component/prometheus.jsonnet

```diff
@@ -48,8 +48,8 @@ local prometheus = {
         },
       ],
     },
-    baseImage: params.images.prometheus.image,
-    tag: params.images.prometheus.tag,
+    image: params.images.prometheus.image + ':' + params.images.prometheus.tag,
+    version: std.split(params.images.prometheus.tag, '@')[0],
```

Comment:
Is this actually correct? We're using the Rancher-provided prometheus-operator, so the Prometheus CRD should not change by updating the Prometheus image in this component.

Reply:
> Is this actually correct

Well it did what we expected us to do:
- the correct image & tag is used
- the deprecation message in the prometheus-operator logs were gone

> so the Prometheus CRD should not change by updating the Prometheus image in this component.

I don't understand this part. The component directly manages the Prometheus CRD?

Reply:
> the deprecation message in the prometheus-operator logs were gone

I wasn't aware about deprecation messages in the operator logs.

> I don't understand this part. The component directly manages the Prometheus CRD?

The component manages a Prometheus CR, not the CRD.

Reply:
Ah, now I understand: We (by "accident") discovered the following message in the cattle-prometheus-operator logs:

    level=warn ts=..., field spec.baseImage is deprecated, 'spec.image' field should be used instead"
    level=warn ts=..., field spec.tag is deprecated, 'spec.image' field should be used instead"

Both the old and new fields still exist in the CRD, so we can use either in the CR.